script/list-page.js

```diff
@@ -0,0 +1,205 @@
+const sidebar = document.querySelector("#sidebar-product");
+const listpage = document.querySelector("#list");
+const gridpage = document.querySelector("#grid");
+const show = document.querySelector("#show");
+const paginationID = document.querySelector("#pagination");
+let limit = 9;
+function start() {
+  getAllProduct((products) => pagination(products));
+  getCategories((categories) => renderSidebar(categories, "danh mục sản phẩm"));
+  getCategoriesPrice((categories) => renderSidebar(categories, "tìm theo mức giá"));
+  getCategoriesColor((categories) => renderSidebar(categories, "tìm theo màu sắc"));
+  getProduct((products) => renderProductListPage(products), (page = 1), (limit = 9));
+  getProduct((products) => renderProductGridPage(products), (page = 1), (limit = 18));
+}
+
+start();
+
+// Get API
+function getCategories(callback) {
+  fetch("http://localhost:3000/categories")
+    .then((response) => response.json())
+    .then(callback);
+}
+
+function getCategoriesPrice(callback) {
+  fetch("http://localhost:3000/categoriesPrice")
+    .then((response) => response.json())
+    .then(callback);
+}
+
+function getCategoriesColor(callback) {
+  fetch("http://localhost:3000/categoriesColor")
+    .then((response) => response.json())
+    .then(callback);
+}
+
+function getProduct(callback, page = 1, limit = 9) {
+  fetch(`http://localhost:3000/products?_page=${page}&_limit=${limit}`)
+    .then((response) => response.json())
+    .then(callback);
+}
+
+function getAllProduct(callback) {
+  fetch(`http://localhost:3000/products`)
+    .then((response) => response.json())
+    .then(callback);
+}
+
+//Filter
+function filterShow(event) {
+  var selectElement = event.target;
+  var value = selectElement.value;
```

Comment:
```suggestion
  const { value } = event.target;
  limit = value
  callAPI()
```
Hàm này có mục đích làm gì nhỉ, anh tưởng phải set lại limit rồi call lại API chứ?

Reply:
Cái này em viết hàm filter thử xem nhưng chưa được nên em để vậy, em đã xóa hàm này